meta-resin-pyro/recipes-containers/crun/crun_%.bbappend

```diff
@@ -0,0 +1 @@
+PACKAGECONFIG_remove = "systemd"
```

Comment:
This doesn't seem right as balenaOS is running systemd. What's the reason for this?

Reply:
I seem to remember older Yocto releases have a systemd version that does not work with crun, so DTs with older Yocto releases cannot use crun.
It maybe better just to append the packagegroup in these older layers to remove crun.

Reply:
@alexgg Crun uses systemd for certain features, but expects certain symbols in libsystemd that are not present in the versions provided by pyro and earlier releases. That doesn't mean the runtime doesn't work (though it remains to be tested), but that the features provided by linking to systemd are disabled.

Glancing at the man page and code, it appears that this is primarily used for configuring cgroups through systemd, though cgroups can also be configured directly through the cgroupfs backend.

Reply:
@jakogut unless we can guarantee crun works fine, I'd rather not include it. So it's either preparing and running a comprehensive test plan for older releases or not including it in them.

Reply:
@alexgg What would a comprehensive test plan involve? I've already added a test case for ensuring that crun can be configured as the default runtime, and can run containers on a given release.

Reply:
@jakogut I personally don't like the idea of diverging the OS builds with two different runtimes, so what I am thinking about is in using the best runtime. If that is crun, then let's make it the default.
With that in mind, we are speaking about replacing the container runtime, so it can potentially affects all containers, from the supervisor to user apps. A test plan would need to give us 100% guarantee that there are no unknown problems lurking around. Something like choosing a representative variety of applications and making sure they work on all the different architectures supported, adding some stress testing with a couple of apps, long run testing to make sure it doesn't have long term effects (memory leaks etc), and then carefully staging a release on smaller to bigger fleets to monitor and fix any problem. Something at least as comprehensive as we have done for the overlay migration.
Probably a good idea to merge this with an engine & go update, so that the test plan can be shared.
That's my point of view which I acknowledge tends to be conservative - once we speak about it on an arch call we will have a broader variety of opinions.

Reply:
@jakogut I didn't quite answer the question though - the above is the test plan we would need to make crun the default runtime. Then we need to either repeat it for older releases were we build it differently, or just not replace it on older Yocto releases (which I prefer)